### PR TITLE
Add support for guru as a docs tool

### PR DESCRIPTION
**Pull request by stamblerre**
Support guru as a tool that can be used for jumping to definitions.

### CODE REVIEW

File: src/goDeclaration.ts [hunk 1]
```diff
@@ -156,6 +158,50 @@ function definitionLocation_gogetdoc(document: vscode.TextDocument, position: vs
...
+				definitionInfo.file = match[1];
+				definitionInfo.line = +match[2] - 1;
+				definitionInfo.column = +match[3] - 1;
+				return resolve(definitionInfo);
+			} catch (e) {
+				reject(e);
+			}
+		});
+		let documentText = document.getText();
+		let documentArchive = document.fileName + '\n';
```
Comment (ramya-rao-a) [inline src/goDeclaration.ts]: you can use `getFileArchive()` from `util.ts` to do this

File: src/goDeclaration.ts [hunk 2]
```diff
@@ -39,8 +39,10 @@ export function definitionLocation(document: vscode.TextDocument, position: vsco
 		// Assume it's > Go 1.5
 		if (toolForDocs === 'godoc' || (ver && (ver.major < 1 || (ver.major === 1 && ver.minor < 6)))) {
 			return definitionLocation_godef(document, position, offset, includeDocs);
+		} else if (toolForDocs == 'guru') {
```
Comment (ramya-rao-a) [inline src/goDeclaration.ts]: There are linting errors that is breaking the build. 
- `==` should be `===` in line 42
- There is a trailing space in line 45

File: src/goSignature.ts [hunk 3]
```diff
@@ -43,7 +43,7 @@ export class GoSignatureHelpProvider implements SignatureHelpProvider {
 				let funcName = declarationText.substring(0, nameEnd);
 				sig = declarationText.substring(sigStart);
 				si = new SignatureInformation(funcName + sig, res.doc);
-			} else {
+			} else if (res.toolUsed === 'gogetdoc') {
```
Comment (ramya-rao-a) [inline src/goSignature.ts]: Both Signature Help and Hover Info features are currently broken.
Until `guru` has the feature to provide docs, you can call `godoc` to provide the docs.

Pull https://github.com/Microsoft/vscode-go/blob/1302fcf0f704f66c974e043258b7ed9783ba104b/src/goDeclaration.ts#L85-L105 out into a function and use it in both `definitionLocation_godef` and `definitionLocation_guru`

Comment (stamblerre) [inline src/goSignature.ts]: Latest changes should fix this - if guru is set as the docsTool, I just passed in a config object with "godoc" set as the docsTool for provideHover and provideSignatureHelp.

**Pull request merged**